bool left_ls
bool right_ls
bool middle
float32 bed_length
